frontend/src/pages/Auth/SignUp.tsx

```diff
@@ -77,24 +78,31 @@ export const SignUp: React.FC = () => {
 	const workspaceInvite = data?.workspace_for_invite_link
 
 	const handleSubmit = useCallback(
-		async ({ user }: firebase.auth.UserCredential) => {
-			if (user?.email) {
+		async ({ additionalUserInfo, user }: firebase.auth.UserCredential) => {
+			const isNewUser = additionalUserInfo?.isNewUser && user?.email
+
+			if (isNewUser) {
 				analytics.track('Sign up', {
-					email: user.email,
+					email: user.email!,
+					provider: additionalUserInfo.providerId,
 				})
-			}
 
-			if (!user?.emailVerified) {
-				auth.currentUser?.sendEmailVerification()
-			}
+				if (!user?.emailVerified) {
+					auth.currentUser?.sendEmailVerification()
+				}
 
-			await createAdmin()
-			message.success('Account created succesfully!')
+				await createAdmin()
+				message.success('Account created succesfully!')
+			}
 
+			await fetchAdmin()
```

Comment:
is this what ensures we block other requests until the token is set by the auth.tsx `/login` request?

Reply:
ah awesome, thanks for getting to the bottom of this